pkg/actuators/machine/reconciler.go

```diff
@@ -211,7 +199,7 @@ func (r *Reconciler) exists() (bool, error) {
 	}
 
 	if err != nil {
-		if strings.Contains(err.Error(), "Not Found") {
+		if strings.Contains(err.Error(), "NOT_FOUND") {
```

Comment:
is this checking an HTTP response?

if so, i wonder if we could use the kubernetes error function `IsNotFound`?

reference, https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound